src/main/java/org/alliancegenome/curation_api/jobs/executors/GafExecutor.java

```diff
@@ -0,0 +1,154 @@
+		String orgAbbrev = tok[tok.length - 1].toUpperCase();
+		String abbr = orgAbbrev.split("\\.")[0];
+		Organization organization = organizationService.getByAbbr(abbr).getEntity();
+
+		// curie, List<GO curie>
+		Map<String, List<String>> uiMap = new HashMap<>();
+		Set<String> orgIDs = new HashSet<>();
+		GZIPInputStream stream = new GZIPInputStream(new FileInputStream(bulkLoadFileHistory.getBulkLoadFile().getLocalFilePath()));
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(stream))) {
+			Stream<String> lines = br.lines();
+
+			// Process each line
+			lines.filter(s -> !s.startsWith("!") && StringUtils.isNotEmpty(s)).forEach(s -> {
+				String[] token = s.split("\t");
+				String orgID = token[0];
+				orgIDs.add(orgID);
+				String modID = token[1];
+				String goID = token[4];
+				if (abbr.equals(orgID)) {
+					List<String> goIDs = uiMap.computeIfAbsent(modID, list -> new ArrayList<>());
+					goIDs.add(goID);
+				}
+			});
+
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		String name = bulkLoadFileHistory.getBulkLoad().getName();
+
+		Map<Long, GeneOntologyAnnotationDTO> gafMap = service.getGafMap(organization);
+		List<Long> gafIdsBefore = new ArrayList<>(gafMap.keySet().stream().toList());
+		gafIdsBefore.removeIf(Objects::isNull);
+
+		List<Long> geneGoIdsLoaded = new ArrayList<>();
+		ProcessDisplayHelper ph = new ProcessDisplayHelper();
+		ph.addDisplayHandler(loadProcessDisplayService);
+		List<GeneOntologyAnnotationDTO> dtos = uiMap.entrySet()
+			.stream()
+			.map(entry -> entry.getValue().stream().map(goID -> {
+				GeneOntologyAnnotationDTO dto = new GeneOntologyAnnotationDTO();
+				dto.setGeneIdentifier(abbr + ":" + entry.getKey());
+				dto.setGoTermCurie(goID);
+				return dto;
+			}).toList()).flatMap(Collection::stream).toList();
+
+		ph.startProcess(name, dtos.size());
+		dtos.forEach(modID -> {
+			Long geneID = service.getGeneID(modID, abbr);
+			if (geneID != null) {
+				GeneOntologyAnnotation newGaf = service.insert(modID, abbr).getEntity();
+				if (newGaf != null) {
+					geneGoIdsLoaded.add(newGaf.getId());
+					bulkLoadFileHistory.incrementCompleted();
+				} else {
+					bulkLoadFileHistory.incrementSkipped();
+				}
+			} else {
+				addException(bulkLoadFileHistory, new ObjectUpdateException.ObjectUpdateExceptionData(modID, "Could not find gene " + modID.getGeneIdentifier(), null));
+				bulkLoadFileHistory.incrementFailed();
+			}
+			ph.progressProcess();
+		});
+		bulkLoadFileHistory.setTotalCount(dtos.size());
+		runCleanupSpecial(service, bulkLoadFileHistory, abbr, gafIdsBefore, geneGoIdsLoaded, "GAF Load");
+		ph.finishProcess();
+		updateHistory(bulkLoadFileHistory);
+
+		bulkLoadFileHistory.finishLoad();
+		updateHistory(bulkLoadFileHistory);
+		updateExceptions(bulkLoadFileHistory);
+	}
+
+	private void runCleanupSpecial(GafService service, BulkLoadFileHistory history, String abbr, List<Long> gafIdsBefore, List<Long> geneGoIdsLoaded, String gafLoad) {
```

Comment:
Why is this method special? I can't see any difference between this and the generic runCleanup method.  If it really is special, no need to pass in the GafService as it's injected.

Reply:
I removed the specialized cleanup code as it finally got obsoleted.